lib/MusicBrainz/Server/Data/Search.pm

```diff
@@ -520,10 +520,18 @@ sub schema_fixup
                 name => delete $data->{status}
             )
         }
+
+        my $packaging = delete $data->{packaging};
+
         if ($data->{packaging}) {
-            $data->{packaging} = MusicBrainz::Server::Entity::ReleasePackaging->new(
-                name => delete $data->{packaging}
-            )
+            $data->{packaging} = ($packaging->{name} && $packaging->{id})
```

Comment:
If `$packaging` can still be a string you may want to check `ref($packaging) eq 'HASH'` before these key accesses, otherwise Perl might break with "can't use string as a hash ref."

Reply:
Thank you, fixed in https://github.com/metabrainz/musicbrainz-server/pull/1231/commits/a42067b0fe8c616bd3164d9d04e302050b4aaf8a#diff-273b5368fc014db576aec607d9a783aaR527.